android/src/test/java/com/rudderstack/android/internal/infrastructure/LifecycleObserverPluginTest.kt

```diff
@@ -1,11 +1,86 @@
 package com.rudderstack.android.internal.infrastructure
 
+import com.rudderstack.android.ConfigurationAndroid
+import com.rudderstack.android.storage.AndroidStorage
+import com.rudderstack.android.utils.TestExecutor
+import com.rudderstack.android.utils.busyWait
 import com.rudderstack.core.Analytics
+import com.rudderstack.core.ConfigDownloadService
+import com.rudderstack.core.Configuration
+import com.rudderstack.core.Logger
+import com.rudderstack.core.RudderUtils
+import com.rudderstack.core.Storage
+import com.rudderstack.core.internal.KotlinLogger
+import com.rudderstack.gsonrudderadapter.GsonAdapter
+import com.rudderstack.jacksonrudderadapter.JacksonAdapter
+import com.rudderstack.models.Message
+import com.rudderstack.models.RudderServerConfig
+import com.rudderstack.models.TrackMessage
+import com.rudderstack.models.TrackProperties
+import com.rudderstack.moshirudderadapter.MoshiAdapter
+import com.rudderstack.rudderjsonadapter.JsonAdapter
+import com.vagabond.testcommon.Verification
+import com.vagabond.testcommon.assertArgument
+import com.vagabond.testcommon.generateTestAnalytics
 import io.mockk.mockk
 import io.mockk.verify
+import junit.framework.TestSuite
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.aMapWithSize
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.any
+import org.hamcrest.Matchers.hasEntry
+import org.hamcrest.Matchers.hasProperty
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.isA
+import org.hamcrest.Matchers.not
+import org.junit.After
+import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Suite
+import org.mockito.ArgumentMatchers
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
 
-class LifecycleObserverPluginTest {
+abstract class LifecycleObserverPluginTest {
+    private lateinit var analytics: Analytics
+    private lateinit var storage: AndroidStorage
```

Comment:
Please rename it to `mockStorage`, as it is is being used to store mock object and I've noticed that we are using the `mock` as a prefix for storing mock objects e.g., mockControlPlane. Therefore, I think it would be better if we follow the same convention for this variable also.